Lib/test/test_asyncio/test_streams.py

```diff
@@ -1129,6 +1129,29 @@ async def inner(httpd):
 
         self.assertEqual(messages, [])
 
+    def test_unclosed_server_resource_warnings(self):
+        async def inner(rd, wr):
+            with self.assertWarns(ResourceWarning) as cm:
+                del wr
+                gc.collect()
+                self.assertEqual(len(cm.warnings), 1)
+                self.assertTrue(str(cm.warnings[0].message).startswith("unclosed <StreamWriter"))
```

Comment:
Shouldn't this be outside the `with`?

Reply:
That doesn't seem to be necessary. It triggers properly with the old code.

I copied the style from the existing `ResourceWarnings` tests to be consistent.

Reply:
Okay!